lib/preview/video.js

```diff
@@ -0,0 +1,149 @@
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+var exec = require('child_process').exec;
+var fs = require('fs');
+var path = require('path');
+var util = require('util');
+
+var PreviewConstants = require('./constants');
+var PreviewImage = require('./image');
+var Result = require('./model').Result;
+
+/**
+ * Process a video file
+ *
+ * @param  {Context}    ctx                 The job context
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Object}     callback.err        An error object, if any
+ * @param  {Result}     callback.result     The result for the video file
+ */
+var process = module.exports.process = function(ctx, callback) {
+  extractFirstFrame(ctx, function(err, imagePath) {
+    if (err) {
+      return callback(err);
+    }
+
+    PreviewImage.processImage(ctx, imagePath, function(err, imageResult) {
+      if (err) {
+        return callback(err);
+      }
+
+      convertVideoIfNecessary(ctx, function(err, convertedVideoPath) {
+        if (err) {
+          return callback(err);
+        }
+
+        if (convertedVideoPath) {
+          imageResult.metadata.converted_video = convertedVideoPath;
+        }
+        var result = new Result(PreviewConstants.STATUS.DONE, imageResult.thumbnail, imageResult.image, null, imageResult.metadata);
+
+        return callback(null, result);
+      });
+    });
+  });
+};
+
+/**
+ * Extract the first frame as a PNG image
+ *
+ * @param  {Context}    ctx                          The job context
+ * @param  {Function}   callback                     Standard callback function
+ * @param  {Object}     callback.err                 An error object, if any
+ * @param  {String}     callback.imagePath           The path to the extracted image
+ */
+var extractFirstFrame = function(ctx, callback) {
+  var extractCommand = [
+    'ffmpeg',
+    util.format('-i "%s"', ctx.sourcePath),
+    '-vframes 1',
+    path.join(ctx.directory, 'frame%d.png')
+  ].join(' ');
+
+  exec(extractCommand, {'timeout': 200000}, function(err, stdout, stderr) {
+    if (err) {
+      var msg = util.format('Error while extracting first frame image from video: %s', err.message);
+      return callback({'code': 500, 'msg': msg});
+    }
+
+    var imagePath = path.join(ctx.directory, 'frame1.png');
+    if (!fs.existsSync(imagePath)) {
+      return callback({'code': 500, 'msg': 'An image file could not be generated'});
```

Comment:
@paulkerschen One more q: would `imagePath` be useful in msg above? and `convertedVideoPath` at line 143, below? Sorry for nit-pickiness. :)